XYZDrivers/src/main/java/com/xyzdrivers/services/SQLService.java

```diff
@@ -12,41 +12,43 @@
 
 import java.sql.*;
 import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import com.xyzdrivers.data.ConnectionProvider;
 
+@RequestScoped
 public class SQLService
 {
-//variables
-    private Connection DB;
+//variables    
     private DatabaseMetaData DBMetaData;
-    
     private PreparedStatement statement;
     private ResultSetMetaData resultsMetaData;
     private ResultSet results;
-//constructors
-    /**
-     * Constructor for jdbcDriver.
-     * <code>dbConnection</code> is tested by retrieving <code>DatabaseMetaData</code> from the <code>Connection</code>.
-     * 
-     * @param dbConnection A <code>Connection</code> to the DB being accessed.
-     * 
-     * @throws SQLException
-     */
-    public SQLService(Connection dbConnection)
-            throws SQLException
+    
+    @Inject
+    private ConnectionProvider connectionProvider;
+    
+    @PostConstruct
+    private void postConstruct()
     {
-        //pass DB connection
-        DB = dbConnection;
-        
-        //get database metadata
-        DBMetaData = DB.getMetaData();
+        try {
```

Comment:
What does this do? Seems to assign a unused variable

Reply:
You're right, I didn't even check if it was used!

Reply:
I deleted my last comment, this is actually used by getPrimaryKeyName and getAllTablesNames - which are functions that need fixing anyway.